packages/salesforce-adapter/src/filters/currency_iso_code.ts

```diff
@@ -0,0 +1,114 @@
+* you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+import _ from 'lodash'
+import { Element, isObjectType, ObjectType, ElemID, ListType, InstanceElement, CORE_ANNOTATIONS, ReferenceExpression } from '@salto-io/adapter-api'
+import Joi from 'joi'
+import { FilterWith } from '../filter'
+import { SALESFORCE, FIELD_ANNOTATIONS, RECORDS_PATH, SETTINGS_PATH, CUSTOM_VALUE, CURRENCY_CODE_TYPE_NAME } from '../constants'
+import { Types, getTypePath } from '../transformers/transformer'
+
+const currencyCodeType = new ObjectType(
+  {
+    elemID: new ElemID(SALESFORCE, CURRENCY_CODE_TYPE_NAME),
+    fields: {
+      [FIELD_ANNOTATIONS.VALUE_SET]: { refType: new ListType(Types.valueSetType) },
+    },
+    annotations: {
+      [CORE_ANNOTATIONS.CREATABLE]: false,
+      [CORE_ANNOTATIONS.DELETABLE]: false,
+      [CORE_ANNOTATIONS.UPDATABLE]: false,
+    },
+    isSettings: true,
+    path: getTypePath(CURRENCY_CODE_TYPE_NAME),
+  }
+)
+
+const CURRENCY_CODE_FIELD_NAME = 'CurrencyIsoCode'
+
+type ValueSet = {}
+
+type CurrencyIsoCodeType = ObjectType & {
+  fields: {
+    [CURRENCY_CODE_FIELD_NAME]: {
+      annotations: {
+        valueSet: ValueSet[]
+        valueSetName?: ReferenceExpression
+      }
+    }
+  }
+}
+
+const VALUE_SET_SCHEMA = Joi.object({
+  valueSet: Joi.array().items({
+    [CUSTOM_VALUE.FULL_NAME]: Joi.string().required(),
+    [CUSTOM_VALUE.DEFAULT]: Joi.boolean().required(),
+    [CUSTOM_VALUE.LABEL]: Joi.string().required(),
+    [CUSTOM_VALUE.IS_ACTIVE]: Joi.boolean().required(),
+  }).required(),
+}).unknown(true).required()
+
+const isTypeWithCurrencyIsoCode = (elem: ObjectType): elem is CurrencyIsoCodeType => {
+  if (!_.has(elem.fields, CURRENCY_CODE_FIELD_NAME)) {
+    return false
+  }
+  const { error } = VALUE_SET_SCHEMA.validate(elem.fields[CURRENCY_CODE_FIELD_NAME]?.annotations)
+  return error === undefined
+}
+
+const transformCurrencyIsoCodes = (
+  element: CurrencyIsoCodeType,
+  currencyCodeInstance: InstanceElement,
+): void => {
+  const currencyIsoCodesRef = new ReferenceExpression(
+    currencyCodeInstance.elemID,
+    currencyCodeInstance
+  )
+
+  delete element.fields.CurrencyIsoCode.annotations.valueSet
+  element.fields.CurrencyIsoCode.annotations.valueSetName = currencyIsoCodesRef
+}
+
+const createCurrencyCodesInstance = (supportedCurrencies: ValueSet): InstanceElement => (
+  new InstanceElement(
+    ElemID.CONFIG_NAME,
+    currencyCodeType,
+    { [FIELD_ANNOTATIONS.VALUE_SET]: supportedCurrencies || [] },
```

Comment:
do we need the `||` case? seems like the type says it cannot be undefined...
anyway, if we do need it, then we should use the `??` operator
```suggestion
    { [FIELD_ANNOTATIONS.VALUE_SET]: supportedCurrencies ?? [] },
```